composer.json

```diff
@@ -20,7 +20,18 @@
 		"johnbillion/query-monitor": "^3.7.1",
 		"altis/dev-tools-command": "^0.4.0",
 		"wp-phpunit/wp-phpunit": "5.8.1",
-		"phpunit/phpunit": "^7.5.20"
+		"phpunit/phpunit": "^7.5.20",
+		"lucatume/wp-browser": "^3.0",
+		"codeception/module-asserts": "^1.0",
+		"codeception/module-phpbrowser": "^1.0",
+		"codeception/module-webdriver": "^1.0",
+		"codeception/module-db": "^1.0",
+		"codeception/module-filesystem": "^1.0",
+		"codeception/module-cli": "^1.0",
+		"codeception/util-universalframework": "^1.0"
+	},
+	"replace": {
+		"ext-zip": "*"
```

Comment:
Not had to use `replace` before, what effect does this have?

Reply:
It prevents composer from checking if the dependency is available, since that extension is only used within the web container, not on the host ( where composer itself runs ). Otherwise composer will block the install.

Reply:
Ah - so I have the `zip` extension installed with my host version of PHP and this causes it to break on composer update with the following:

```
Problem 1
    - altis/dev-tools[dev-master, dev-codeception] cannot be installed as that would require removing ext-zip[1.15.6]. altis/dev-tools replaces ext-zip and thus cannot coexist with it.
    - ext-zip is present at version 1.15.6 and cannot be modified by Composer
    - Root composer.json requires altis/dev-tools * -> satisfiable by altis/dev-tools[dev-codeception, dev-master].

To enable extensions, verify that they are enabled in your .ini files:
    - /usr/local/etc/php/7.4/php.ini
    - /usr/local/etc/php/7.4/conf.d/ext-opcache.ini
    - /usr/local/etc/php/7.4/conf.d/local-dev.ini
You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
```

Might do a quick poll to see how common it is to have that extension or not. Got a few ways forward:

- see how hard it'd be to require everyone to install the zip extension
- see if we could circumvent that requirement somehow in the core composer plugin
- see if we can force composer commands to run on the PHP container somehow

Reply:
Oooh, I didn't anticipate that issue, makes sense now that I see the error!

I'd probably skip option #3 about running composer inside the container, seems like the most troublesome. Will check if there is another way to ignore that with composer config, before asking everyone to install the zip extension.